notebooks/MAST/HSC/CASJOBS/HSCV3_SMC_API/hscv3_smc_api.ipynb

```diff
@@ -0,0 +1,516 @@
+    "    \n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    table (string): summary, detailed, propermotions, or sourcepositions\n",
+    "    release (string): v3 or v2\n",
+    "    magtype (string): magaper2 or magauto (only applies to summary table)\n",
+    "    baseurl: base URL for the request\n",
+    "    \n",
+    "    Returns a string with the base URL for this request\n",
+    "    \"\"\"\n",
+    "    checklegal(table, release, magtype)\n",
+    "    if table == \"summary\":\n",
+    "        url = f\"{baseurl}/{release}/{table}/{magtype}\"\n",
+    "    else:\n",
+    "        url = f\"{baseurl}/{release}/{table}\"\n",
+    "    return url\n",
+    "\n",
+    "\n",
+    "def checklegal(table, release, magtype):\n",
+    "    \"\"\"Checks if this combination of table, release and magtype is acceptable\n",
+    "    \n",
+    "    Raises a ValueError exception if there is problem\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    releaselist = (\"v2\", \"v3\")\n",
+    "    if release not in releaselist:\n",
+    "        raise ValueError(f\"Bad value for release (must be one of {', '.join(releaselist)})\")\n",
+    "    if release == \"v2\":\n",
+    "        tablelist = (\"summary\", \"detailed\")\n",
+    "    else:\n",
+    "        tablelist = (\"summary\", \"detailed\", \"propermotions\", \"sourcepositions\")\n",
+    "    if table not in tablelist:\n",
+    "        raise ValueError(f\"Bad value for table (for {release} must be one of {', '.join(tablelist)})\")\n",
+    "    if table == \"summary\":\n",
+    "        magtypelist = (\"magaper2\", \"magauto\")\n",
+    "        if magtype not in magtypelist:\n",
+    "            raise ValueError(f\"Bad value for magtype (must be one of {', '.join(magtypelist)})\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Helper function for crowded scatterplots\n",
+    "\n",
+    "`select_subset()` selects a sample of points in crowded regions to speed up plotting. This converts the fastkde PDF values (which are probabilities per unit area using the x and y data units) to the expected number of points within a region covered by the marker on the plot.  It then picks only a few points to plot in crowded regions while selecting all the points in uncrowded regions.\n",
+    "\n",
+    "Inputs: \n",
+    "* `zs` = array of PDF density values computed using fastkde\n",
+    "* `axes` = axis range tuple from fastkde\n",
+    "* `markersize` = matplotlib marker size (which is an area in points)\n",
+    "* `over` = oversampling factor (empirically 5 fills in the holes)\n",
+    "* `verbose` = print additional info\n",
+    "\n",
+    "Returns an array of subscripts to plot from `zs`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def select_subset(zs,axes,markersize=2,over=5,verbose=False):\n",
```

Comment:
```suggestion
    "def select_subset(zs, axes, markersize=2, over=5, verbose=False):\n",
```
(minor PEP8 quibble)

Reply:
Thanks, I must of missed that one when doing the flake8 updates. 

Done